cmd/ndt7-client/main.go

```diff
@@ -183,10 +191,25 @@ func main() {
 		InsecureSkipVerify: *flagNoVerify,
 	}
 	r.client.FQDN = *flagHostname
-	if *flagBatch {
-		r.emitter = emitter.NewBatch()
+
+	var e emitter.Emitter
+
+	// If -batch, force -format=json.
+	if *flagBatch || *flagFormat == "json" {
```

Comment:
I don't understand what this code does? Are you checking whether `*flagBatch != nil`?

Reply:
(I think we can use `flagx` to specify beforehand the available values and then `switch` here. This would strike me as the most @m-lab way of solving this kind of problem; feel free to open an issue if you'd rather not do that now?)

Reply:
*flagBatch is `-batch`. It cannot be nil as it has a default value of `false`.

If either one of:
1. `-batch` 
2. `-format=json`

have been passed to the program, the output format will be json. The goal is to handle the deprecation of `-batch` gracefully, in case someone is depending on it. Does it answer your question?

I would not change the flag parsing package here as it feels completely out of the scope for this PR.